src/sink/wavefront.rs

```diff
@@ -285,13 +378,98 @@ impl Sink for Wavefront {
 
 #[cfg(test)]
 mod test {
+    use quickcheck::{QuickCheck, TestResult};
     use super::*;
     use chrono::{TimeZone, Utc};
     use metric::{TagMap, Telemetry};
     use sink::Sink;
     use std::sync::Arc;
 
     #[test]
+    fn test_no_unpadded_gaps() {
+        fn inner(bin_width: u8, ms: Vec<Telemetry>) -> TestResult {
+            if bin_width == 0 {
+                return TestResult::discard();
+            }
+            // TODO
+            //
+            // The fact that I have to set a bucket here suggests that 'padding'
+            // might be sensible for inclusion as an interator on the
+            // Bucket. But! It doesn't have to be that way, if the
+            // Vec<Telemetry> is pre-prepped.
+            let mut bucket = Buckets::new(bin_width as i64);
+            for m in ms.clone() {
+                bucket.add(m);
+            }
+            let mut padding = padding(bucket.into_iter(), bin_width as i64).peekable();
+
+            while let Some(t) = padding.next() {
+                if let Some(next_t) = padding.peek() {
+                    // When we examine the next point in a series there are
+                    // three possibilities:
+                    //
+                    //  1. the points don't hash the same, so we move on
+                    //  2. the points do hash the same:
+                    //     a. if their timestamps are greater than one span
+                    //        apart then they are both zero
+                    //     b. if both points are non-zero they must not be
+                    //        more than one span apart
+                    if t.hash() == next_t.hash() {
+                        let span = (t.timestamp - next_t.timestamp).abs() / (bin_width as i64);
```

Comment:
Ah! Found it in the nomicon, [here](https://doc.rust-lang.org/nomicon/casts.html).

> For numeric casts, there are quite a few cases to consider:
> 
> * casting between two integers of the same size (e.g. i32 -> u32) is a no-op
> * casting from a larger integer to a smaller integer (e.g. u32 -> u8) will truncate
> * casting from a smaller integer to a larger integer (e.g. u8 -> u32) will
> * zero-extend if the source is unsigned
> * sign-extend if the source is signed
> * casting from a float to an integer will round the float towards zero
>     * NOTE: currently this will cause Undefined Behavior if the rounded value cannot be represented by the target integer type. This includes Inf and NaN. This is a bug and will be fixed.
> * casting from an integer to float will produce the floating point representation of the integer, rounded if necessary (rounding strategy unspecified)
> * casting from an f32 to an f64 is perfect and lossless
> * casting from an f64 to an f32 will produce the closest possible value (rounding strategy unspecified)
>     * NOTE: currently this will cause Undefined Behavior if the value is finite but larger or smaller than the largest or smallest finite value representable by f32. This is a bug and will be fixed.